oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OAuth2TokenIssuer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+
+/**
+ * Base interface for token issuer implementations. Implementations can chose to return any type of an OAuth2 token,
+ * {@code TOKEN_TYPE} which implements {@code AbstractOAuth2Token}.
+ *
+ * @see JwtAccessTokenOAuth2TokenIssuer
+ * @see OpaqueRefreshTokenOAuth2TokenIssuer
+ *
+ * @author Alexey Nesterov
+ * @since 0.0.4
+ */
+public interface OAuth2TokenIssuer<TOKEN_TYPE extends AbstractOAuth2Token> {
```

Comment:
Spring Security doesn't use named generic types so please change `TOKEN_TYPE` to `T`